io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -464,6 +464,89 @@ impl Lvol {
                 }),
         }
     }
+    /// List All Snapshot.
+    pub async fn list_all_snapshots() -> Vec<VolumeSnapshotDescriptor> {
+        let mut snapshot_list: Vec<VolumeSnapshotDescriptor> = Vec::new();
+        let bdev =
+            UntypedBdev::bdev_first().expect("Failed to enumerate devices");
+
+        let lvol_devices = bdev
+            .into_iter()
+            .filter(|b| b.driver() == "lvol")
+            .map(|b| Lvol::try_from(b).expect("Can't create Lvol from device"))
+            .collect::<Vec<Lvol>>();
+        // return empty snapshot parameter list, if no snapshot found.
+        if lvol_devices.len() <= 1 {
+            return snapshot_list;
+        }
+        for lvol in lvol_devices {
+            let blob = lvol.bs_iter_first();
+            let mut skip_lvol = false;
+            let mut snapshot_param: SnapshotParams = Default::default();
+            for attr in SnapshotXattrs::iter() {
+                let mut val = std::ptr::null();
+                let mut size: u64 = 0;
+                let attr_id = attr.name().to_string().into_cstring();
+                let curr_attr_val;
+                unsafe {
+                    let r = spdk_blob_get_xattr_value(
+                        blob,
+                        attr_id.as_ptr(),
+                        &mut val as *mut *const c_void,
+                        &mut size as *mut u64,
+                    );
+                    // skip current lvol if any of the snapshot attribute not
+                    // found.
+                    if r != 0 {
```

Comment:
Could we add a warning message here to capture the fact that some attributes are missing ?

Reply:
> Could we add a warning message here to capture the fact that some attributes are missing ?

Accepted. Modified Code.